### PR TITLE
PR7 - Remove AlicaEngine::this from teammanager

### PR DESCRIPTION
**Pull request by triccyx**
Remove AlicaEngin::this from teammanager and subclasses

### CODE REVIEW

**Comment by veerajsk** [comment]
@triccyx needs rebase

File: alica_engine/include/engine/collections/RobotEngineData.h [hunk 1]
```diff
@@ -10,14 +10,16 @@
 namespace alica
 {
 class AlicaEngine;
```
Comment (christianhelm) [inline alica_engine/include/engine/collections/RobotEngineData.h]: Declaration of AlicaEngine can be removed now

File: alica_engine/include/engine/collections/RobotProperties.h [hunk 2]
```diff
@@ -12,12 +12,13 @@ namespace alica
 {
 
 class AlicaEngine;
```
Comment (christianhelm) [inline alica_engine/include/engine/collections/RobotProperties.h]: Declaration of AlicaEngine can be removed now

File: alica_engine/include/engine/collections/SuccessMarks.h [hunk 3]
```diff
@@ -7,6 +7,7 @@ namespace alica
 {
 
 class AlicaEngine;
```
Comment (christianhelm) [inline alica_engine/include/engine/collections/SuccessMarks.h]: Can be removed

File: alica_engine/include/engine/teammanager/Agent.h [hunk 4]
```diff
@@ -19,6 +19,7 @@ class AbstractPlan;
 class EntryPoint;
 class DomainVariable;
 class SuccessMarks;
+class PlanRepository;
```
Comment (christianhelm) [inline alica_engine/include/engine/teammanager/Agent.h:21]: AlicaEngine can be removed

File: alica_engine/include/engine/util/idFunctions.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (christianhelm) [inline alica_engine/include/engine/util/idFunctions.h]: All other file names start with a capital letter so this one should too. E.g. IdFunctions.h

File: alica_engine/include/engine/collections/RobotProperties.h (hunk 2 above)
Comment (veerajsk) [reply alica_engine/include/engine/collections/RobotProperties.h]: +1

File: alica_engine/include/engine/teammanager/Agent.h [hunk 6]
```diff
@@ -43,13 +44,15 @@ class Agent
     void setTimeout(AlicaTime t);
 
 private:
-    Agent(const AlicaEngine* engine, AlicaTime timeout, const std::string& defaultRole, const AgentAnnouncement& aa);
+    Agent(const ModelManager& modelManager, const PlanRepository& planRepository, const AlicaClock& clock, AlicaTime timeout, const std::string& defaultRole,
+            const AgentAnnouncement& aa);
 
     const AlicaEngine* _engine;
```
Comment (veerajsk) [inline alica_engine/include/engine/teammanager/Agent.h]: this engine reference needs to be removed

File: alica_engine/include/engine/util/idFunctions.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ * This method can be used, e.g., for passing a part of a ROS
+ * message and receiving a pointer to a corresponding Identifier object.
+ */
+
+/**
+ * Generates random ID.
+ * @return The ID
+ */
+
+inline alica::AgentId GenerateID()
```
Comment (veerajsk) [inline alica_engine/include/engine/util/idFunctions.h]: can simply make `generateID` a private static method of the TeamManager class.

File: alica_engine/include/engine/util/idFunctions.h [hunk 8]
```diff
@@ -0,0 +1,28 @@
+
+#pragma once
+
+#include "engine/Types.h"
+#include <chrono>
+#include <random>
+
+/**
+ * If present, returns the ID corresponding to the given prototype.
```
Comment (veerajsk) [inline alica_engine/include/engine/util/idFunctions.h]: this comment can be removed

File: alica_engine/include/engine/collections/RobotEngineData.h (hunk 1 above)
Comment (triccyx) [reply alica_engine/include/engine/collections/RobotEngineData.h]: ok

File: alica_engine/include/engine/collections/SuccessMarks.h (hunk 3 above)
Comment (triccyx) [reply alica_engine/include/engine/collections/SuccessMarks.h]: ok

File: alica_engine/include/engine/teammanager/Agent.h (hunk 4 above)
Comment (triccyx) [reply alica_engine/include/engine/teammanager/Agent.h:21]: ok

File: alica_engine/include/engine/teammanager/Agent.h (hunk 6 above)
Comment (triccyx) [reply alica_engine/include/engine/teammanager/Agent.h]: ok

File: alica_engine/include/engine/util/idFunctions.h (hunk 8 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: ok

File: alica_engine/include/engine/util/idFunctions.h (hunk 7 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: ok

File: alica_engine/include/engine/util/idFunctions.h (hunk 5 above)
Comment (triccyx) [reply alica_engine/include/engine/util/idFunctions.h]: File removed

File: alica_engine/include/engine/collections/RobotProperties.h (hunk 2 above)
Comment (triccyx) [reply alica_engine/include/engine/collections/RobotProperties.h]: ok

**Review by christianhelm — APPROVED** [review]

File: alica_engine/src/engine/collections/RobotEngineData.cpp [hunk 9]
```diff
@@ -19,8 +19,9 @@ namespace alica
 /**
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/collections/RobotEngineData.cpp:18]: Engine include can be removed.

File: alica_engine/src/engine/collections/RobotProperties.cpp [hunk 10]
```diff
@@ -6,12 +6,13 @@
 namespace alica
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/collections/RobotProperties.cpp:5]: Engine include can be removed

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 11]
```diff
@@ -3,11 +3,13 @@
 #include "engine/AlicaEngine.h"
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/teammanager/TeamManager.cpp]: You can remove the engine include and add the following includes instead:

#include "engine/IAlicaCommunication.h"
#include "engine/ConfigChangeListener.h"
#include "engine/PlanRepository.h"

File: alica_engine/src/engine/teammanager/TeamManager.cpp [hunk 12]
```diff
@@ -53,17 +55,27 @@ bool AgentsCache::addAgent(Agent* agent)
...
 }
 
-TeamManager::TeamManager(AlicaEngine* engine, AgentId agentID)
+TeamManager::TeamManager(ConfigChangeListener& configChangeListener, const ModelManager& modelManager, const PlanRepository& planRepository,
+        const IAlicaCommunication& communicator, const AlicaClock& clock, Logger& log, int version, uint64_t masterPlanId, const std::string& localAgentName,
+        AgentId agentID)
         : _localAgent(nullptr)
-        , _localAgentID(agentID)
-        , _engine(engine)
+        , _configChangeListener(configChangeListener)
```
Comment (bjoernschroeder) [inline alica_engine/src/engine/teammanager/TeamManager.cpp]: The configChangeListener is only used in the constructor, we can use the passed instance directly and not store it locally.

File: alica_engine/src/engine/collections/RobotEngineData.cpp (hunk 9 above)
Comment (triccyx) [reply alica_engine/src/engine/collections/RobotEngineData.cpp:18]: ok

File: alica_engine/src/engine/collections/RobotProperties.cpp (hunk 10 above)
Comment (triccyx) [reply alica_engine/src/engine/collections/RobotProperties.cpp:5]: ok

File: alica_engine/src/engine/teammanager/TeamManager.cpp (hunk 11 above)
Comment (triccyx) [reply alica_engine/src/engine/teammanager/TeamManager.cpp]: ok

File: alica_engine/src/engine/teammanager/TeamManager.cpp (hunk 12 above)
Comment (triccyx) [reply alica_engine/src/engine/teammanager/TeamManager.cpp]: ok

**Comment by triccyx** [comment]
I think we can merge

**Comment by veerajsk** [comment]
@triccyx the code gen jar files have gotten into the PR, please remove them, otherwise the PR looks fine

**Review by veerajsk — CHANGES_REQUESTED** [review]
please remove the jar files from the PR

**Comment by triccyx** [comment]
files removed

**Review by veerajsk — APPROVED** [review]

**Pull request merged**